File: web-frontend/src/domain/entities/Ad.ts
export interface Ad {
  id: string;
  type: AdType;
  title: string;
  description: string;
  reward: string;
  image: string;
  example: string;
  requirements?: string[];
  duration?: number; // seconds
  maxRewards?: number;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

export type AdType =
  | "expose"        // 노출형
  | "sns"           // SNS형
  | "ua"            // UA 특화형
  | "app_inflow"    // 앱 유입형
  | "action"        // 액션 유도형
  | "live"          // 라이브커머스형
  | "retargeting"   // 리타겟팅형
  | "ranking";      // 랭킹 부스팅형

export interface User {
  id: string;
  email: string;
  name: string;
  totalRewards: number;
  completedAds: string[];
  createdAt: Date;
}

export interface Reward {
  id: string;
  adId: string;
  userId: string;
  amount: number;
  status: 'pending' | 'completed' | 'failed';
  claimedAt?: Date;
  createdAt: Date;
}

File: android-sdk/samplesdk/src/main/java/com/example/android_sdk/domain/entities/Ad.kt
package com.example.android_sdk.domain.entities

data class Ad(
    val id: String,
    val type: AdType,
    val title: String,
    val description: String,
    val reward: String,
    val image: String,
    val example: String,
    val requirements: List<String>? = null,
    val duration: Int? = null,
    val maxRewards: Int? = null,
    val isActive: Boolean = true,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

enum class AdType {
    EXPOSE, SNS, UA, APP_INFLOW, ACTION, LIVE, RETARGETING, RANKING
}

data class User(
    val id: String,
    val email: String,
    val name: String,
    val totalRewards: Int,
    val completedAds: List<String>,
    val createdAt: Long = System.currentTimeMillis()
)

data class Reward(
    val id: String,
    val adId: String,
    val userId: String,
    val amount: Int,
    val status: RewardStatus,
    val claimedAt: Long? = null,
    val createdAt: Long = System.currentTimeMillis()
)

enum class RewardStatus {
    PENDING, COMPLETED, FAILED
}
